storage/remote/wal_watcher.go

```diff
@@ -0,0 +1,362 @@
+// Copyright 2018 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/tsdb"
+	"github.com/prometheus/tsdb/wal"
+	fsnotify "gopkg.in/fsnotify/fsnotify.v1"
+)
+
+var (
+	segmentRegex              = regexp.MustCompile(`\d{8}`)
+	checkpointRegex           = regexp.MustCompile(`checkpoint.\d{6}`)
+	watcherSamplesRecordsRead = prometheus.NewGaugeVec(
```

Comment:
This sounds like a counter

Reply:
@brian-brazil these metrics seem more useful to me as gauges that reset to 0 when prometheus is restarted or the remote write config is applied again. Let me know if you disagree.

Reply:
though they should at least be of the same type as the samples in and samples succeeded metrics, those are both counters atm